docs/rationale.md

```diff
@@ -23,4 +23,72 @@ try (Scope scope = TracingContextUtils.currentContextWith(span)) {
 ```
 
 It would not be possible to call `recordException` if `span` was also using try-with-resources.
-Because this is a common usage for spans, we do not support try-with-resources.
\ No newline at end of file
+Because this is a common usage for spans, we do not support try-with-resources.
+
+
+## Versioning and Releases
+
+### Assumptions
+
+- This project uses semver v2, as does the rest of OpenTelemetry.
+
+### Goals
+
+- API Stability: 
+    - Once the API for a given signal (spans, logs, metrics, baggage) has been officially released, code instrumented with that API module will 
+function, *with no recompilation required*, with any API+SDK that has the same major version, and equal or greater minor or patch version. 
+    - For example, libraries that are instrumented with `opentelemetry-api-trace:1.0.1` will function, at runtime with 
+SDK library `opentelemetry-sdk-trace:1.11.33` plus `opentelemetry-api-trace:1.11.33`. 
```

Comment:
But we planned to version SDK independently from API, right?

Reply:
I think this depended on how hard we go on preserving SDK compatibility. The more precise statement is indeed probably the use of the BOM for the latest version rather than mentioning them explicitly.

Reply:
I can update to refer to the BOM. Whether we actually end up versioning the SDK independently from the API remains to be seen (I sure hope we don't have to).

Reply:
done